alerting/src/test/kotlin/org/opensearch/alerting/util/clustermetricscoordinatortest/ClusterMetricsCoordinatorIT.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.util.clustermetricscoordinatortest
+
+import org.apache.http.entity.ContentType
+import org.apache.http.entity.StringEntity
+import org.junit.After
+import org.junit.Before
+import org.opensearch.alerting.AlertingRestTestCase
+import org.opensearch.alerting.makeRequest
+import org.opensearch.alerting.model.ClusterMetricsDataPoint
+import org.opensearch.alerting.opensearchapi.string
+import org.opensearch.alerting.util.ClusterMetricsVisualizationIndex
+import org.opensearch.client.Response
+import org.opensearch.client.ResponseException
+import org.opensearch.common.xcontent.XContentFactory.jsonBuilder
+import org.opensearch.common.xcontent.XContentType
+import org.opensearch.rest.RestStatus
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.util.*
+import kotlin.collections.ArrayList
+class ClusterMetricsCoordinatorIT : AlertingRestTestCase() {
+    @Before
+    fun setup() {
+        // When setting up the tests, change the execution frequency and history max age settings to 1 minute and 10 minutes from
+        // 15 minutes and 7 days.
+        generateData()
+        val response = getSettings()
+        val persistentMap = response["persistent"] as Map<String, Any>
+        val executionFrequency = persistentMap["plugins.alerting.cluster_metrics.execution_frequency"].toString()
+        val storageTime = persistentMap["plugins.alerting.cluster_metrics.metrics_history_max_age"].toString()
+        assertEquals(executionFrequency, "1m")
+        assertEquals(storageTime, "10m")
+    }
+
+    fun `test checkName`() {
+        // WHEN + THEN, check whether the created index exists and has the name '.opendistro-alerting-cluster-metrics'
+        val index = ClusterMetricsVisualizationIndex.CLUSTER_METRIC_VISUALIZATION_INDEX
+        val response = client().makeRequest("HEAD", index)
+        assertEquals("Index $index does not exist.", RestStatus.OK, response.restStatus())
+    }
+
+    fun `test numberDocs`() {
+        // Check that the total number of documents found is divisible by the total number of metric types.
```

Comment:
Same comment as above; the test name can consist of essentially what is in this comment.